monitoring/prober/rid/test_isa_expiry.py

```diff
@@ -7,14 +7,29 @@
 from . import common
 from .common import SCOPE_READ, SCOPE_WRITE
 
+ISA_ID = '00000098-ba6d-4c20-a575-6e412e000000'
```

Comment:
I would prefer if we introduced a pool of well-known test IDs here, and pick an ID at random at the beginning of each test run (for all parts that need an ID).

Reply:
What I'm thinking we get from defining them individual in files is that if someone sees, e.g., "00000098-ba6d-4c20-a575-6e412e000000" somewhere in their database, they can quickly search the codebase and find out what test caused the resource with that test ID to be injected.  If we choose randomly from a pool of well-known IDs, it would be harder to track down the reason an artifact was left behind in the database.  I agree that always using the same IDs doesn't exercise the system quite as much (we would never have discovered #314, for instance), but it seems like the tradeoff is worth it.  What were the benefits you wanted to capture from having a pool of well-known test IDs?

Reply:
I think maintaining a pool of random IDs would allow for running multiple prober test instances in parallel. With only one ID, we would run the risk of clashing tests (not the end of the world). More to this argument, some sort of `fuzzing` usually helps a lot in uncovering more elaborate issues faster (as you pointed out already).

I feel like maintaining a set of IDs (in a file, one ID per line) would enable both the tests and GC-infrastructure to use the IDs and semi-atuomatically clean up, too, even in the case of failing tests.

Reply:
Could you elaborate on how the one-per-line IDs in a file would make multiple prober test instances in parallel possible?  I could imagine a function that generates a random UUID and then zeroes the first and last 6 digits, but that wouldn't need a file and would be prone to permanently leaving artifacts behind (even upon a subsequent successful test run).  If a file were used, we would need a lot of IDs in it to avoid the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem) unless IDs were drawn sequentially, which would disable multiple parallel test instances.  With that many IDs, I'm not sure it would be reasonable to attempt to clean them all up on every test run.  Also, the method of cleanup is dependent on the object type represented by the ID, so cleaning up everything at the very end would necessitate keeping a type registry which I'm not sure how we'd do.

It seems like putting test sequences in test classes would be the best intermediate suggestion, as test classes can have a destructor that cleans up after themselves.  But I'm still not super excited about that approach with random UUIDs because that will leave behind artifacts that won't be cleaned up next test run if something (like the cleanup phase) fails.

Here are benefits I think we may be interested in:
1. Make sure a successful test suite run cleans any artifacts from previously-failed test suite runs
1. Be able to identify which test(s) did not clean up an artifact
1. Be able to differentiate test resource IDs from real resource IDs at a glance
1. Run multiple prober instances in parallel
1. Explore a wide space of UUID values
1. Clean up artifacts by the end of the test suite run

I think the current PR approach achieves 1, 2, & 3, does not achieve 4 & 5, and achieves 6 as well as is reasonably possible.  I'm not sure how drawing IDs from a fixed pool (like a file containing a list of IDs) achieves any of the above except 3, and maybe 5 slightly (but not much) better than the current PR approach.  Drawing IDs from a function (like one that generates a random ID and then zeros the first and last 6 digits) achieves 3, 4, and 5, and 6 in the same way as the current PR, but does not achieve 1 or 2.  I think 1 & 2 are currently more valuable than 4 & 5 which is why I have the PR as-is rather than drawing from a function.  I think I don't yet fully understand the pool-of-IDs approach as my visualization of it doesn't seem to check as many boxes as either alternative.

Perhaps it might be best to consider a different method of supplying IDs in a follow-up PR?

Reply:
Works for me.